src/qutip_qip/qiskit/converter.py

```diff
@@ -0,0 +1,160 @@
+from qutip_qip.circuit import QubitCircuit
+from qiskit.quantum_info import Operator
+import numpy as np
+from qutip import Qobj
+import warnings
+
+from qiskit.circuit import QuantumCircuit, Instruction
+
+_map_gates = {
+    "p": "PHASEGATE",
+    "x": "X",
+    "y": "Y",
+    "z": "Z",
+    "h": "SNOT",
+    "s": "S",
+    "t": "T",
+    "rx": "RX",
+    "ry": "RY",
+    "rz": "RZ",
+    "swap": "SWAP",
+}
+
+_map_controlled_gates = {
+    "cx": "CX",
+    "cy": "CY",
+    "cz": "CZ",
+    "crx": "CRX",
+    "cry": "CRY",
+    "crz": "CRZ",
+    "cp": "CPHASE",
+}
+
+_ignore_gates = ["id", "barrier"]
+
+
+def _make_user_gate(unitary, instruction: Instruction):
+    """
+    Returns a user defined gate from a unitary matrix.
+
+    Parameters
+    ----------
+    unitary : numpy.array
+        The unitary matrix describing the gate.
+
+    instruction : qiskit.circuit.Instruction
+        Qiskit instruction containing info about the gate.
+    """
+
+    def user_gate():
+        return Qobj(
+            unitary,
+            dims=[[2] * instruction.num_qubits, [2] * instruction.num_qubits],
+        )
+
+    return user_gate
+
+
+def _get_qutip_index(qiskit_index: int, total_qubits: int) -> int:
+    """
+    Map the bit index from qiskit to qutip.
+
+    When we convert a circuit from qiskit to qutip,
+    the 1st bit is mapped to the nth bit and so on.
+    """
+    return total_qubits - 1 - qiskit_index
+
+
+def qiskit_to_qutip(qiskit_circuit: QuantumCircuit) -> QubitCircuit:
+    """
+    Convert a QuantumCircuit from qiskit to qutip_qip's QubitCircuit.
+
+    Parameters
+    ----------
+    qiskit_circuit : QuantumCircuit
+        QuantumCircuit to be converted to QubitCircuit. 
+
+    Returns
+    -------
+    qutip_circuit : QubitCircuit
+        The converted circuit in qutip_qip's QubitCircuit format.
+    """
+    qubit_map = {}
+    for qiskit_index, qubit in enumerate(qiskit_circuit.qubits):
+        qubit_map[qubit] = _get_qutip_index(
+            qiskit_index, qiskit_circuit.num_qubits
+        )
+
+    clbit_map = {}
+    for qiskit_index, clbit in enumerate(qiskit_circuit.clbits):
+        clbit_map[clbit] = _get_qutip_index(
+            qiskit_index, qiskit_circuit.num_clbits
+        )
```

Comment:
We can probably update `_get_qutip_index` such that if the input is a list of integers, it iterates over it and calls itself.

It also has a few use cases later. So that you don't have to use `[qubit_map[qreg] for qreg in qiskit_qregs]`.